catalog/ui/catalog-ui-search/src/main/webapp/component/search-form/search-form.collection.js

```diff
@@ -0,0 +1,50 @@
+/**
```

Comment:
We should refactor this to make it so the non-custom (predefined) templates can be changed to immediately even though the custom ones are still loading.  

We should consider if we want to keep the results of the get request around so that we don't have to refetch them every time the search form selector is closed and reopened.

Reply:
refactored to return non-custom templates immediately

Reply:
Is it worth considering caching custom templates in local storage to improve load time? Might improve user experience.